codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -54,7 +54,11 @@
  */
 public class DefaultHttp2Connection implements Http2Connection {
 
-    private final Set<Listener> listeners = new HashSet<Listener>(4);
+    /**
+     * We chose a {@link List} over a {@link Set} to avoid allocating an {@link Iterator} objects when iterating over
+     * the listeners.
+     */
+    private final List<Listener> listeners = new ArrayList<Listener>(4);
```

Comment:
Seems like I made a good choice then ;)



> Am 03.04.2015 um 23:45 schrieb Nathan Mittler <notifications@github.com>:
> 
> In codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java:
> 
> > @@ -54,7 +54,11 @@
> >   */
> >  public class DefaultHttp2Connection implements Http2Connection {
> >  
> > -    private final Set<Listener> listeners = new HashSet<Listener>(4);
> > +    /**
> > +     * We chose a {@link List} over a {@link Set} to avoid allocating an {@link Iterator} objects when iterating over
> > +     * the listeners.
> > +     */
> > +    private final List<Listener> listeners = new ArrayList<Listener>(4);
> @normanmaurer I'm pretty sure that 4 was your suggestion originally :)
> 
> —
> Reply to this email directly or view it on GitHub.
>